src/ui/layout/component.tsx

```diff
@@ -0,0 +1,89 @@
+import classNames from "classnames";
+import React, { Component, ReactElement } from "react";
+import {
+  AlignContent,
+  AlignItems,
+  Display,
+  FlexDirection,
+  FlexWrap,
+  JustifyContent,
+  Position,
+} from "./models";
+import "./style.scss";
+
+interface PublicProps {
+  className?: string;
+  padding?: Padding;
+  margin?: Margin;
+  display?: Display;
+  flexDirection?: FlexDirection;
+  flexWrap?: FlexWrap;
+  justifyContent?: JustifyContent;
+  alignItems?: AlignItems;
+  alignContent?: AlignContent;
+  position?: Position;
+}
+
+interface Spacing {
+  y?: size;
+  x?: size;
+  t?: size;
+  b?: size;
+  r?: size;
+  l?: size;
+  [key: string]: size | undefined;
+}
+
+type size = 1 | 2 | 3 | 4 | 5;
+type Padding = Spacing;
+type Margin = Spacing;
+type Props = PublicProps;
+
+class Layout extends Component<Props> {
```

Comment:
super nit: this has no state. should it just be a SFC then?

Reply:
my React is rusty here, but an SFC component wouldn't allow me to have private methods right? Should I use a `PureComponent` instead?

Reply:
private functions are just typescript sugar anyhoo. you can define "private" functions outside the stateless component and just not export them outside the file. if you want to be funky or have them to reference locally scoped variables, you can define functions inside the component:
```
export function MyComponent() {
  const thing = "stuff";
  function innerFunc() {
    // do stuff with {thing}
  }

  return <OtherComponent prop={innerFunc()}/>
}
```